sysmontask/proc.py

```diff
@@ -277,12 +285,12 @@ def procInit(self):
     for pi in pids:
         procs=ps.Process(pi)
 
-        if(procs.username()!='root'):
-            if procs.name()=='systemd':
-                self.systemdId.append(pi)
-                self.processSystemd.append(procs)
-                searcher(self,procs,None)          ## for multiple user view
-                # break
+        # if(procs.username()!='root'):
+        if procs.name()=='systemd':
+            self.systemdId.append(pi)
+            self.processSystemd.append(procs)
+            searcher(self,procs,None)          ## for multiple user view
+            # break
```

Comment:
removing procs.username()!=root: just gets all the process starting from the process PID 1(or 2)(which is the systemd process for root) those process does not need to show in the user process. For gnome type desktop environments the process starting from user's systemd are the process which needs to be seen(the user process).

Reply:
from my latop(Linux Mint 19.3 Cinnamon) the ps tree:
![ps tree](https://i.imgur.com/XdHUCem.png)
and the owner of PID 1 is root. without the process, this process tree will be empty:
![w/o pid 1](https://i.imgur.com/yEUyBE3.png)

Reply:
yes, you are right but as of now, the process tab works well only for gnome desktop environments.
The thing is in Linux-mint-cinnamon the user process is categorized under lightdm->cinnamon launcher on the other hand in the gnome desktop environment they come under the user's systemd.  
In just a couple of days, other desktop env will also be supported. If you interested in making it work for Linux-mint-cinnamon we can talk about details.

Reply:
else:
Can remove that part and then I can merge for compatibility with python3<=3.7.

Reply:
hello, i push new code to check platform is Mint as below:
`        
(dist,ver,nname) = platform.dist()
`
`        
if(procs.username()!='root' or dist =="LinuxMint"):
`

Reply:
Hi, the platform.dist() has been deprecated starting from python3.8 so it works only till 3.7. The alternative approach that we can do is to check the desktop environments.

Reply:
Hi, sorry for the late. i build two distributions, Debian 9 stretch and Ubuntu 20 focal, to watch systemd behaviors and rewrote the codes to get right systemd process.